src/app/host/[room]/page.tsx

```diff
@@ -1,46 +1,150 @@
-import HostControls from '@/components/HostControls'
 import RoomMainUi from '@/components/RoomMainUi'
+import RoomInfo from '@/components/RoomInfo'
 
-export default function HostRooms({ params }: { params: { room: string } }) {
-	const { room } = params
+export default function HostRoom({ params }: { params: { room: string } }) {
+	const [allUserPointData, setAllUserPointData] = useState<ListenerRes[]>([])
+	const storyPointRes = useSocketListener('story-points')
+	const joinRoomRes = useSocketListener('join-room')
 
-	console.log('%c>>> HOST ROOM roomId:', 'color: #5f0', room)
+	useEffect(() => {
+		if (!storyPointRes) return
+
+		console.log('%c>>> listener', 'color: red', storyPointRes)
+		setAllUserPointData((prevUsersPoints) => {
+			const index = prevUsersPoints.findIndex((data) => {
+				return data.userName === storyPointRes.userName
+			})
+			if (index !== -1) {
+				const noDuplicates = [...prevUsersPoints]
+				noDuplicates[index].message = storyPointRes.message
+				console.log('%c>>> noDuplicates SP', 'color: red', noDuplicates)
+				return noDuplicates
+			}
+			return [...prevUsersPoints, storyPointRes]
+		})
+	}, [storyPointRes])
+
+	useEffect(() => {
+		if (!joinRoomRes) return
+
+		console.log('%c>>> joinRoomRes', 'color: red', joinRoomRes)
+
+		setAllUserPointData((prevUsersPoints) => {
+			const index = prevUsersPoints.findIndex((data) => {
+				return data.userName === joinRoomRes.userName
+			})
+			if (index !== -1) {
+				const noDuplicates = [...prevUsersPoints]
+				noDuplicates[index].message = joinRoomRes.message
+				console.log('%c>>> noDuplicates JR', 'color: red', noDuplicates)
+				return noDuplicates
+			}
+			return [...prevUsersPoints, joinRoomRes]
+		})
+	}, [joinRoomRes])
+
+	const { room } = params
 
 	const hostDataLocalStorage = localStorage.getItem('scrumDivingHostData')
-	const hostData = hostDataLocalStorage && JSON.parse(hostDataLocalStorage)
+	const hostName =
+		hostDataLocalStorage && JSON.parse(hostDataLocalStorage)?.hostName
 
-	console.log(
-		'%c>>> HOST ROOM hostData from LocalStorage',
-		'color: yellow',
-		hostData,
-	)
-	const hostFirstName = hostData?.hostName
+	useEffect(() => {
+		const timeStamp = Date.now().toString()
+		socketRoomEmitter('join-room', 'join', hostName, timeStamp, room)
+	}, [room, hostName])
 
 	useEffect(() => {
-		socketRoomEmitter('join-room', 'Room Joined', room, hostFirstName)
-	}, [room, hostFirstName])
+		console.log(
+			'%c>>> allUserPointData Host Room',
+			'color: #f0f',
+			allUserPointData,
+		)
+		const timeStamp = Date.now().toString()
+		socketRoomEmitter(
+			'all-users-story-points',
+			allUserPointData,
+			hostName,
+			timeStamp,
+			room,
+		)
+		localStorage.setItem(
+			'scrumDivingStoryPoints',
+			JSON.stringify([allUserPointData]),
+		)
+	}, [allUserPointData, room, hostName])
+
+	const handleShowPoints = () => {
+		const timeStamp = Date.now().toString()
+		socketRoomEmitter(
+			'show-disable-reset-points',
```

Comment:
suggestion: This event name is so confusing 😅  It holds a boolean value (true/false), and it's about showing the points. So here's a suggestion: `isPointsVisible`

Reply:
This one name / emitter is used in 3 different components to trigger 3 different actions. 

When true, it shows points and disable story point buttons. 

When set back to false, it resets points, enables story point buttons, and hides points. So it is named for exactly what it does: show-disable-reset-points

Reply:
I understood that after a long debugging around. It's just not easy to understand. This is another place where we can simplify the flow and reduce the amount of states. But it's okay, we'll get to this part again later on after the useEffects :)

Reply:
this effects 2 states that occur in different components, show-disable. The event though is emitted on show points and reset points, show-reset. So however you try to shorten it, something gets left out. That's why it is so long.

Reply:
Then maybe, we need 2 events: show-points and reset-points, and which one has their own effects. Something for you to consider later on if you feel the need.

Reply:
Okay, will do. I'm going to mark this resolved.